litestar/contrib/repository/abc/_async.py

```diff
@@ -253,27 +253,31 @@ def check_not_found(item_or_none: T | None) -> T:
         return item_or_none
 
     @classmethod
-    def get_id_attribute_value(cls, item: T | type[T]) -> Any:
+    def get_id_attribute_value(cls, item: T | type[T], id_attribute: Any | None = None) -> Any:
         """Get value of attribute named as :attr:`id_attribute <AbstractAsyncRepository.id_attribute>` on ``item``.
 
         Args:
             item: Anything that should have an attribute named as :attr:`id_attribute <AbstractAsyncRepository.id_attribute>` value.
+            id_attribute: Allows customization of the unique identifier to use for model fetching.
+                Defaults to `None`, but can reference any surrogate or candidate key for the table.
 
         Returns:
             The value of attribute on ``item`` named as :attr:`id_attribute <AbstractAsyncRepository.id_attribute>`.
         """
-        return getattr(item, cls.id_attribute)
+        return getattr(item, id_attribute if id_attribute is not None else cls.id_attribute)
 
     @classmethod
-    def set_id_attribute_value(cls, item_id: Any, item: T) -> T:
+    def set_id_attribute_value(cls, item_id: Any, item: T, id_attribute: Any | None = None) -> T:
```

Comment:
```suggestion
    def set_id_attribute_value(cls, item_id: Any, item: T, id_attribute: str | None = None) -> T:
```

Any reason not to do this?

Reply:
I don't think so.  I left it off since it sets it for the entire class, but I think it could be helpful in certain cases.

Reply:
No, str would be just fine here.  I'll go back and update.